transport_nantes/mobilito/views.py

```diff
@@ -436,3 +452,50 @@ def get(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
         context = self.get_context_data(**kwargs)
         self.check_view_permission(context["mobilito_session"])
         return self.render_to_response(context)
+
+
+@csrf_protect
+def report_session(request: HttpRequest, **kwargs) -> HttpResponse:
+    """Report a session as inappropriate"""
+
+    if request.method == 'POST':
+        if request.user.is_authenticated:
+            user = request.user
+        else:
+            user = None
+
+        session_sha1 = kwargs['session_sha1']
+        try:
+            # In the event that somehow, in spite of CSRF protection, a user
+            # tries to report a session that doesn't exist, we don't want to
+            # crash the app, so we catch the exception, log it and return a 404
+            session_object = get_object_or_404(
+                Session, session_sha1=session_sha1)
+        except Http404:
+            logger.error(f"{user or 'Anonymous User'} tried to report a "
+                         "non-existing session")
+            raise Http404
+        if session_object:
+            client_ip, _ = get_client_ip(request)
+            _, created = InappropriateFlag.objects.get_or_create(
+                session=session_object,
+                reporter_user=user,
+                reporter_tn_session_id=request.session.get('tn_session'),
+                defaults={
+                    'session': session_object,
+                    'reporter_user': user,
+                    'reporter_tn_session_id': request.session.get('tn_session'),
+                    'report_details': request.POST.get('report-abuse-text'),
+                    'reporter_ip_address': client_ip,
+                })
+            if created:
+                logger.info(f"{user or 'Anonymous User'} reported session "
+                            f"{session_object.id}")
+            else:
+                logger.info(f"{user or 'Anonymous User'} tried to report session "
```

Comment:
Language: are these flags or reports?

Reply:
If we can tell that the person has already flagged this session, then why did we permit them to do so?

Also, again, log enough that you can debug if you need to.  You are teasing your future self by not leaving useful information in the log, only a T/F.